docs/reference/hooks.rst

```diff
@@ -566,12 +566,33 @@ Hooks for customising the way users are directed through the process of creating
 
   Modify the final list of action menu items on the page creation and edit views. The callable passed to this hook receives a list of ``ActionMenuItem`` objects, a request object and a context dictionary as per ``register_page_action_menu_item``, and should modify the list of menu items in-place.
 
+
   .. code-block:: python
 
     @hooks.register('construct_page_action_menu')
     def remove_submit_to_moderator_option(menu_items, request, context):
         menu_items[:] = [item for item in menu_items if item.name != 'action-submit']
 
+
+.. _construct_page_action_menu:
+
+  Can also be used to customize default action menu button. The last item in menu_item variable is chosen as the default action. An example on changing the default to publish can be seen below.
+
+  .. code-block:: python
+
+    from wagtail.admin.action_menu import UnpublishMenuItem, DeleteMenuItem, SubmitForModerationMenuItem, SaveDraftMenuItem, PublishMenuItem
+
+    @hooks.register('construct_page_action_menu')
+    def make_publish_default_action(menu_items, request, context):
+        menu_items[:] = [
+                UnpublishMenuItem(order=10),
+                DeleteMenuItem(order=20),
+                SubmitForModerationMenuItem(order=30),
+                SaveDraftMenuItem(order=40),
+                PublishMenuItem(order=50),
+        ]
```

Comment:
It would be good to update this example so that it doesn't require redefining the whole list. Perhaps something like:

    @hooks.register('construct_page_action_menu')
    def make_publish_default_action(menu_items, request, context):
        for item in menu_items:
            if isinstance(item, PublishMenuItem):
                item.order = 100

(although changing the `order` on an existing menu item feels a bit sneaky, and I'm not 100% sure we should encourage it...)

Reply:
Can I just add a line to the comment saying, "redefining the whole list is not necessary, but for the purposes of a clear example, the list is redefined to show why PublishMenuItem is default. " That way, the example is neither sneaky nor misleading.

Reply:
I've realised that my suggested code won't work, because `construct_page_action_menu` is called after the list has been sorted by `item.order`, and so at that point the list's actual order is meaningful rather than the `order` attribute. I think the simplest / clearest example for making Publish the default would therefore be:

    @hooks.register('construct_page_action_menu')
    def make_publish_default_action(menu_items, request, context):
        for (index, item) in menu_items:
            if item.name == 'action-publish':
                # move to end of list
                menu_items.pop(index)
                menu_items.append(item)
                break

(The fix at https://github.com/wagtail/wagtail/pull/5500#discussion_r332783437 is a good illustration of why we shouldn't encourage redefining the whole list - anyone with that code in place would now need to update it with the new Page Locked item.)

Reply:
Example code updated in https://github.com/gasman/wagtail/commit/6d6bda072462d6a1c859efc45aa0c3181feedcc2